### PR TITLE
[13.0][FIX] purchase_order_product_recommendation_brand: There is not products when a brand is selected

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT29448

ping @chienandalu @CarlosRoca13

### CODE REVIEW

File: purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py [hunk 1]
```diff
@@ -14,8 +14,11 @@ def _get_products(self):
         # Filter products by brand if set.
         # It will apply to show_all_partner_products as well
         if self.product_brand_ids:
+            # We are in onchange context, so to avoid to compare records with
+            # new_id instances we use self.product_brand_ids.ids,
+            # self.product_brand_ids = newid but self.product_brand_ids.ids is a real id
             products = products.filtered(
-                lambda x: x.product_brand_id in self.product_brand_ids
+                lambda x: x.product_brand_id.id in self.product_brand_ids.ids
```
Comment (chienandalu) [inline purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:21]: Maybe better something like this?:

```suggestion
                lambda x: x.product_brand_id._origin in self.product_brand_ids
```

Comment (sergio-teruel) [reply purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:21]: @CarlosRoca13 please, can you explain to @chienandalu why is better use .ids instead origin.
Anyway, the newid instance is self.product_brand_ids not x.product_brand_id

Comment (CarlosRoca13) [reply purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:21]: The field `ids` stores the ids involved on the NewId object. So it is a direct way to get the value. We have to use the field `_origin` when we have a cache record, this normally has `ids = []`

Comment (chienandalu) [reply purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:21]: ok, ok. Tests are failing anyway

**Comment by CarlosRoca13** [comment]
I fix the test error on https://github.com/OCA/purchase-workflow/pull/1163

**Comment by sergio-teruel** [comment]
@chienandalu @CarlosRoca13 Rebase done after https://github.com/OCA/purchase-workflow/pull/1163 has been merged!

**Review by CarlosRoca13 — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Pull request merged**